common/table.js

```diff
@@ -56,15 +56,54 @@
      *      });
      */
     .factory('recordTableUtils', ['DataUtils', '$timeout', function(DataUtils, $timeout) {
-        function read(scope) {
+
+        // This method sets backgroundSearch states depending upon various parameters
+        // If it returns true then we should render the data
+        // else we should reject the data
+        function setSearchStates(scope, isBackground, searchTerm) {
+
+            // If request is background
+            if (isBackground) {
+                // If there is a term in search queue for background and there is no foreground search going on then
+                // Fire the request for the term in the queue and return false
+                // Else empty the queue and set backgroundSearch false
+                if (scope.vm.backgroundSearchQueue && !scope.vm.foregroundSearch) {
+                    scope.vm.search = scope.vm.backgroundSearchQueue
```

Comment:
`backgroundSearchQueue` technically is not a _Queue_ so let's not call it that. It's confusing. I gather from reading the code that it is the _pending_ background search term or something like that. Might rename it `backgroundPendingSearchTerm` or something.